src/Disciplina/Witness/Mode.hs

```diff
@@ -0,0 +1,60 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | Module contains the definition of Witness's WorkMode and its implementations.
+
+module Disciplina.Witness.Mode
+    (
+      -- * Constraints
+      WitnessWorkMode
+
+      -- * Implementations
+    , WitnessContext (..)
+    , WitnessRealMode
+    , wcDB
+    , wcLoggerName
+    ) where
+
+import Universum
+
+import Control.Lens (makeLenses)
+import Ether.Internal (HasLens (..))
+import System.Wlog (HasLoggerName (..), LoggerName)
+
+import Disciplina.DB.Class (MonadDB)
+import Disciplina.DB.Real.Types (NodeDB)
+import qualified Disciplina.Launcher.Mode as Basic
+
+---------------------------------------------------------------------
+-- WorkMode class
+---------------------------------------------------------------------
+
+-- | Set of typeclasses which define capabilities of Witness node.
+type WitnessWorkMode m =
+    ( Basic.BasicWorkMode m
+    , MonadDB m
+    )
+
+---------------------------------------------------------------------
+-- WorkMode implementation
+---------------------------------------------------------------------
+
+data WitnessContext = WitnessContext
+    { _wcDB         :: NodeDB
+    , _wcLoggerName :: LoggerName
+    }
+
+makeLenses ''WitnessContext
+
+type WitnessRealMode = ReaderT WitnessContext IO
+
+---------------------------------------------------------------------
+-- Instances
+---------------------------------------------------------------------
+
+instance HasLens NodeDB WitnessContext NodeDB where
+    lensOf = wcDB
+
+instance {-# OVERLAPPING #-} HasLoggerName WitnessRealMode where
+    askLoggerName = view wcLoggerName
+    modifyLoggerName name = local $ wcLoggerName %~ name
```

Comment:
Do `HasLens` and `HasLoggerName` imply `WithLogger` and `MonadDB` constraints? I didn't see any default instances for these (maybe I'm blind, dunno)

Reply:
I seems so, it's written by you: https://github.com/DisciplinaOU/disciplina/blob/martoon%2Fdscp106-true-modes/src/Disciplina/DB/Real/Functions.hs#L70

Reply:
:thinking: :thinking: :thinking: